### ISSUE TITLE
Permissive file reading

### ISSUE THREAD

**Issue by mgrazebrook** [python-excel/xlrd#164]
Python 3 throws exceptions if you encounter characters in the wrong encoding, e.g. a latin1 copyright symbol in an allegedly utf8 file. 

It would be nice to be able to set the 'errors' parameter of the open() function through xlrd.open_workbook - either as an optional parameter or by passing an open file instead of a file name.

**Comment by cjw296** [comment]
Can you load the file as a string and process it that way?

**Comment by mgrazebrook** [comment]
open_workbook takes a file name not an open file or string so I don't see how. It opens in binary mode (I was just looking at the source) so I imagine my idea of using open's error function would not work the way I imagined it.

**Comment by cjw296** [comment]
Pass file_contents:
https://github.com/python-excel/xlrd/blob/master/xlrd/__init__.py#L385

**Comment by mgrazebrook** [comment]
Good idea. Thanks. 

On closer examination the exception was not in fact thrown by xlrd so this is my mistake. It's from the xlsx to csv program I just posted in answer to someone else's request (and which I'm updating). The script had processed quite a lot of files before I found my bug.

The exception is raised in cp1252.py through csv.writer's writerow(out_row). 

Apologies for taking up your time. Very impressed by your quick response!
